stdlib/src/builtin/reversed.mojo

```diff
@@ -173,3 +164,21 @@ fn reversed[
     return _DictEntryIter[K, V, dict_origin, False](
         src[]._reserved() - 1, 0, src
     )
+
+
+@always_inline
+fn reversed[
```

Comment:
**Suggestion** Should we add some unit tests to ensure this works as expected?

Reply:
Modified the `String`, `StringLiteral`, and `StringSlice` tests to use the new builtins